### DISCUSSION TITLE
Unable to get Japanese file name correctly

### DISCUSSION THREAD

**Discussion by Ezekiel157** [aclap-dev/video-downloadhelper#671] [Help and Q&A]
### Your question

After updating the version of Video DownloadHelper, certain symbols disappear from Japanese file names.

For example, if I download a video from

https://www.youtube.com/watch?v=00Ha5YEsU30

the original file name is

" やき゛さんゆうひ゛ん The Goat's Letters - by チャラン・ポ・ランタン [歌ってみたよ175] "

However, the actually generated file name is

"やきさんゆうひん The Goats Letters - by チャランホランタン 歌ってみたよ175"

It may be difficult for non-Japanese people to understand, but the symbol at the top right of the letters and the dots that separate the characters are gone.

ぎ → き
チャラン・ポ・ランタン → チャランホランタン

Thank you for your correction.

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: ja
coapp: "checking"
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0
{
"used_sidebar_button": true,
"used_popup_button": true,
"download_directory": "F:\",
"never_show_no_incognito_msg_again": true,
"view_options": {
"all_tabs": false,
"low_quality": false,
"sort_by_status": true,
"sort_reverse": false,
"show_button_clean": true,
"show_button_clean_all": false,
"show_button_convert_local": false,
"hide_downloaded": false
},
"use_sidebar": false,
"open_count_store": 5
}

**Comment by GrampaWildWilly** [comment]
You might want to keep an eye on #655.

**Reply by Musouheki** [reply]
Japanese and Korean in Unicode are completely different issues and cannot be used as references.

**Reply by c29979** [reply]
Although the symptoms are different, they should be the identical issue. They both started to happen at the same time, and they both appear to be related to the Unicode normalization NFD/NFKD. The missing dakuten and han-dakuten should be the consequence of a combination of Unicode normalization and the dropping of punctuations.

**Reply by maboroshin** [reply]
Here 671 is first. Immediately after, there is a duplicate report in #738. And it was mentioned Korean character issue: https://github.com/aclap-dev/video-downloadhelper/discussions/655#discussioncomment-9785535

**Comment by paulrouget** [comment]
Hello. That's something I'm working on at the moment, will fix this soon I hope!

**Reply by Musouheki** [reply]
I'm using this Version 9.0.2.12 (stable), target mozilla, locale ja. 
When dealing with Japanese filenames, consider NFD/NFC issues. This is due to the fact that Apple and others have different choices of Unicode normalization methods. Specifically, the kana of the downloaded Japanese file name is missing turbid and semi-turbid sound symbols.
This is probably because my environment is Windows and the server is Windows or Linux, so the app is made in the Apple environment and does not use the convert method.

**Reply by maboroshin** [reply]
Version 9.3.0.1, WIndows 10, FireFox, This problem is ongoing. This also occurs when the following pages are retrieved with the smart naming feature.

1. Open URL: https://pixabay.com/ja/videos/%E3%82%AB%E3%83%83%E3%83%97%E3%83%AB-%E6%81%8B%E4%BA%BA%E3%81%9F%E3%81%A1-%E6%89%8B-%E6%84%9B-152798/
2. Click Test button.
3. The first four letters:
* ```カップル``` : Original string
* ```カッフル``` : Retrieved result
The upper right circle of the 3rd character is missing.

Is this a CoApp issue or a JavaScript issue?

**Reply by Smashir** [reply]
Hello.
Version 9.5.0.1:
Are you not planning to fix this bug? I have written the fix in the comments below (on May 19). This bug is a BIG problem, like mistaking “Bolliwood” for “Holliwood” in English...

**Comment by Smashir** [comment]
Version 9.5.0.1

In,
background/main.js
and
content2/smartnaming_editor.js

Chenge the following:
.normalize("NFD").replace(/\./gu," ").replace(/[^\p{L}\p{N}\-\s]/ug,"").replace(/-+/gu,"-").replace(/\s+/gu," ").substring(0,t.max_length)

to:
.normalize("NF**C**").**replace(/[\\/:*?"<>|]/gu,"")**.replace(/[^\p{L}\p{N}\-\s]/ug,"").replace(/-+/gu,"-").replace(/\s+/gu," ").**normalize("NFD")**.substring(0,t.max_length)

He suggested this workarround in the following link.

#https://tu2na7.blog.fc2.com/blog-entry-191.html

I sincerely hope that the critical bug in this excellent add-on will be fixed soon.
